src/widgets/modals/LoginModal.js

```diff
@@ -126,7 +126,7 @@ class LoginModal extends React.Component {
 
   onSelectLanguage = ({ item }) => {
     const { settings, changeCurrentLanguage } = this.props;
-    changeCurrentLanguage(UIDatabase.getSetting(SETTINGS_KEYS.CURRENT_LANGUAGE));
+    changeCurrentLanguage(item.code);
```

Comment:
just interested, why is the db entry different to the selected language? why is it not stored in the db when changing the language?

Reply:
Oh, that's a good point, it's the next line that updates the DB with the selection (though abstracted through 'settings'):
`settings.set(SETTINGS_KEYS.CURRENT_LANGUAGE, item.code);`

So we could also just reverse the order so the DB gets updated first before trying to update the app state